pkg/xds/experimental/client/gcp/helpers.go

```diff
@@ -0,0 +1,38 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package gcp
+
+import (
+	"fmt"
+
+	"github.com/google/uuid"
+
+	corepb "github.com/cilium/proxy/go/envoy/config/core/v3"
+)
+
+func Node(projectNum int64, mesh string, locality *corepb.Locality, clientID, userAgent string) *corepb.Node {
+	nodeId := fmt.Sprintf("projects/%d/networks/mesh:%s/nodes/%s", projectNum, mesh, clientID)
+	return &corepb.Node{
+		Id:            nodeId,
+		UserAgentName: userAgent,
+		Locality:      locality,
+	}
+}
+
+func LocalityFromZone(zone string) *corepb.Locality {
+	if len(zone) < 3 {
```

Comment:
I don't understand what is this function doing. Can you explain what is zone here? 
If this function is supposed to check against zone and region the gcp one then I don't think this is the right validation. 
https://cloud.google.com/compute/docs/regions-zones#identifying_a_region_or_zone

I think you should return error if zone does not match the requirements

Reply:
Please add UT for this function

Reply:
Removed GCP-specific code.